viewer/viewer.js

```diff
@@ -4969,138 +4973,606 @@ app.post('/removeTags', logAction(), function(req, res) {
+    hunt.id = result._id;
+    return res.send(JSON.stringify({ success: true, hunt: hunt }));
+  });
+});
 
-  if (req.body.ids) {
-    var ids = queryValueToArray(req.body.ids);
+app.get('/hunt/list', logAction('hunt/list'), function (req, res) {
+  if (Config.get('multiES', false)) { return res.molochError(401, 'Not supported in multies'); }
+  if (!req.user.packetSearch) { return res.molochError(403, 'Need packet search privileges'); }
 
-    sessionsListFromIds(req, ids, ["ta", "tags-term", "node"], function(err, list) {
-      searchAndTagList(req, tags, list, function () {
-        return res.send(JSON.stringify({success: true, text: "Tags added successfully"}));
+  let query = {
+    sort: {},
+    from: parseInt(req.query.start) || 0,
+    size: parseInt(req.query.length) || 10000,
+    query: { bool: { must: [] } }
+  };
+
+  query.sort[req.query.sortField || 'created'] = { order: req.query.desc === 'true' ? 'desc': 'asc'};
+
+  if (req.query.history) { // only get finished jobs
+    query.query.bool.must.push({ term: { status: 'finished' } });
+    if (req.query.searchTerm) { // apply search term
+      query.query.bool.must.push({
+        query_string: {
+          query : req.query.searchTerm,
+          fields: ['name', 'userId']
+        }
       });
+    }
+  } else { // get queued, paused, and running jobs
+    query.from = 0;
+    query.size = 1000;
+    query.query.bool.must.push({ terms: { status: ['queued', 'paused', 'running'] } });
+  }
+
+  if (Config.debug) {
+    console.log('hunt query:', JSON.stringify(query, null, 2));
+  }
+
+  Promise.all([Db.searchHunt(query),
+               Db.numberOfHunts()])
+    .then(([hunts, total]) => {
+      if (hunts.error) { throw hunts.error; }
+
+      var results = { total:hunts.hits.total, results:[] };
+      for (let i = 0, ilen = hunts.hits.hits.length; i < ilen; i++) {
+        var hit = hunts.hits.hits[i];
+        var hunt = hit._source;
+        hunt.id = hit._id;
+        hunt.index = hit._index;
+        results.results.push(hunt);
+      }
+
+      var r = {
+        recordsTotal: total.count,
+        recordsFiltered: results.total,
+        data: results.results
+      };
+
+      res.send(r);
+    }).catch(err => {
+      console.error('ERROR - /hunt/list', err);
+      return res.molochError(500, 'Error retrieving hunts - ' + err);
     });
-  } else {
-    sessionsListFromQuery(req, res, ["ta", "tags-term", "node"], function(err, list) {
-      searchAndTagList(req, tags, list, function () {
-        return res.send(JSON.stringify({success: true, text: "Tags added successfully"}));
-      });
+});
+
+app.delete('/hunt/:id', logAction('hunt/:id'), checkCookieToken, function (req, res) {
+  if (Config.get('multiES', false)) { return res.molochError(401, 'Not supported in multies'); }
+  if (!req.user.packetSearch) { return res.molochError(403, 'Need packet search privileges'); }
+
+  function deleteHunt (id) {
```

Comment:
if the hunt is running should we still allow delete?

Reply:
I make sure that the hunt job process is paused before deletion. And I have a check for processing a hunt that might be deleted. So there are checks to make sure it works without error. However, from a user interaction perspective, maybe you should not be able to delete a running session? Then, a user would have to pause, THEN delete the session?